src/StorageWriter/Router.ts

```diff
@@ -12,49 +11,48 @@ import { ExchangeConfiguration } from './ExchangeConfiguration'
 
 export const isTraceError = anyPass([isNoMoreEntriesException])
 
-@injectable()
-export class Router {
-  private readonly logger: Pino.Logger
-  private readonly messaging: Messaging
-  private readonly claimController: ClaimController
-  private readonly exchange: ExchangeConfiguration
-
-  constructor(
-    @inject('Logger') logger: Pino.Logger,
-    @inject('Messaging') messaging: Messaging,
-    @inject('ClaimController') claimController: ClaimController,
-    @inject('ExchangeConfiguration') exchange: ExchangeConfiguration,
-  ) {
-    this.logger = childWithFileName(logger, __filename)
-    this.messaging = messaging
-    this.claimController = claimController
-    this.exchange = exchange
-  }
+export interface Router {
+  readonly start: () => Promise<void>
+  readonly stop: () => Promise<void>
+}
 
-  async start() {
-    await this.messaging.consume(this.exchange.newClaim, this.onNewClaim)
-    await this.messaging.consume(this.exchange.storageWriterStoreNextClaim, this.onStorageWriterStoreNextClaim)
-  }
+export interface Dependencies {
+  readonly logger: Pino.Logger
+  readonly messaging: Messaging
+  readonly claimController: ClaimController
+}
 
-  async stop() {
-    this.logger.info('Stopping StorageWriter Router...')
-    await this.messaging.stop()
-  }
+export interface Arguments {
+  readonly dependencies: Dependencies
+  readonly exchange: ExchangeConfiguration
+}
+
+export const Router = ({
+  dependencies: {
+    logger,
+    messaging,
+    claimController,
+  },
+  // Instead of receiving exchange configuration, a messaging object
```

Comment:
Agreed, but comments in code tend to be forgotten. Can you remove the comment? I went ahead and added it in https://github.com/poetapp/node/issues/66#issuecomment-444312048 so we don't forget.

Reply:
Cool thx